tests/integration/tables/chassis_info.cpp

```diff
@@ -0,0 +1,45 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed in accordance with the terms specified in
+ *  the LICENSE file found in the root directory of this source tree.
+ */
+
+// Sanity check integration test for chassis_info
+// Spec file: specs/windows/chassis_info.table
+
+#include <osquery/tests/integration/tables/helper.h>
+
+namespace osquery {
+namespace table_tests {
+
+class chassisTest : public testing::Test {
+ protected:
+  void SetUp() override {
+    setUpEnvironment();
+  }
+};
+
+TEST_F(chassisTest, test_sanity) {
+  const QueryData data = execute_query("select * from chassis_info");
+  ASSERT_EQ(data.size(), 1ul);
```

Comment:
Instead of this assert, add a check for if (!data.empty()) then keep the row map and call to validate rows in within this check.

Reply:
Though would this explain why the table is simply missing if I build the daemon anyway? This would explain the failing integration test, but it seems like the table is completely absent from the database.

I did add the table into the CMakeLists.txt file in the tables/windows folder before I built, but am I missing it somewhere else too?

Reply:
Ah yes, it is missing!

Reply:
Though I added it to the Cmakelists in the tables folder and I still find that it's absent in the daemon's virtual table even after verifying it is processed during build without errors.

I seem to be having issues running tests since this afternoon so I haven't been able to run it locally, but I suspect there's a similar issue,  cause I don't see chassis_info listed when I list available tables after building it with it present in the cmake file.

Is there another location I need to add it to besides the integration tests and table files?

Reply:
Oh holy sh*t i was sleeping on the integration test. It's missing from the spec and its also got a handful of errors.

I'll fix that tomorrow and push it. I really wasn't paying attention on that one.

Reply:
Hmm, still missing from the tables if i add it in there. How many cmakelists are there that require this? I don't recall having to do this in the past, so it's new to me.

Reply:
I can try to help! Right, before we did a lot implicitly/automatically and now the build is explicit/manual. The new explicit method is ultimately better (IMO) because the build logic is simpler.

Reply:
I'm still seeing missing on my local build after adding it to cmakelists in 3 places
-Specs
-Integration Tests
-Tables

Are there any other places that need it added to the cmakelists?

Reply:
Ah! Success! I didn't add it properly to specs. Pushing a change to that cmakelist and it should build properly now.